app/page.tsx

```diff
@@ -10,27 +10,52 @@ import movieData from "../lib/top_100_movies.json";
 
 const Home = () => {
   const [searchText, setSearchText] = React.useState("");
+  const [filteredMovieData, setFilteredMovieData] = React.useState(movieData);
 
   const handleSearch = (e: React.FormEvent<HTMLInputElement>) => {
     e.preventDefault();
-    setSearchText(e.currentTarget.value);
+    const newSearchText = e.currentTarget.value;
+    setSearchText(newSearchText);
+    const newFilteredMovieData = movieData.filter((movie) =>
+      movie.title.includes(newSearchText.toLocaleLowerCase())
+    );
+    setFilteredMovieData(newFilteredMovieData);
   };
+
+  const handleUpdateFilter = (filterGenre: string) => {
+    const newFilteredMovieData = movieData.filter((movie) => {
+      return movie.genre.includes(filterGenre);
+    });
+    setFilteredMovieData(newFilteredMovieData);
+  };
+
   return (
     <>
-      <h1 className="text-4xl mb-12">Top 100 Movies</h1>
-      <form className="flex justify-between items-center mb-16">
-        <SearchBar
-          id={"search-movie"}
-          label={"search movie title"}
-          type={"search"}
-          name={"search-movie"}
-          value={searchText}
-          placeholder="Type movie name"
-          onChange={handleSearch}
-        />
-        <FliterBar />
-      </form>
-      <MovieSection movieData={movieData} />
+        <h1 className="text-4xl mb-12">Top 100 Movies</h1>
+        <form className="flex justify-between items-center mb-16">
+          <SearchBar
+            id={"search-movie"}
+            label={"search movie title"}
+            type={"search"}
+            name={"search-movie"}
+            value={searchText}
+            placeholder="Type movie name"
+            onChange={handleSearch}
+          />
+          <div className="flex justify-center gap-4">
+            <FliterBar
+              id={"filter-genre"}
+              label={"Genre:"}
```

Comment:
todo: If you pass the label as a prop, then the movies list should be a prop too :)